modules/ting/src/Search/TingSearchResultInterface.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * @file
+ * The TingSearchResultInterface interface.
+ */
+
+namespace Ting\Search;
+
+/**
+ * Interface TingSearchResultInterface
+ *
+ * @package Ting\Search
+ */
+interface TingSearchResultInterface {
```

Comment:
> In the same vain: For now I've allowed some of the open search naming to leak trough, with the intention of refactoring when I get to the actual primo implementation.

That's a valid approach. I prefer to refactor the naming early to get a feel for how it's working and not getting to used to the old. Means I have to refactor if it isn't right, but there's no penalty if I get it right in the first try.

> Sidenote: shouldn't the TingCollection Class in ding which is a subclass of DingEntityCollection have been called TingEntityCollection?

> With regards to getCollections() - both the search-providers and Drupal has the concept of "Collections". The first is wrappers around somewhat raw search-results and the latter is an entity. It's valid to call both "Collections". Which is why I went for being explicit and calling the getter getTingEntityCollections.

Uhm, but there's no TingEntityCollection, and TingEntity doesn't exist, so following your logic, it should be `getTingCollections()`. But please don't name your types in the method name, they're already documented in the dockblock.

Well, no, the idea was that collections was first class entities.